### PR TITLE
Feature/start 40 #1

### PR DESCRIPTION
**Pull request by HerbertIV**
Add report term

### CODE REVIEW

File: src/Http/Requests/ReportTermConsultationRequest.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+namespace EscolaLms\Consultations\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class ReportTermConsultationRequest extends FormRequest
+{
+    public function authorize(): bool
+    {
+        $user = auth()->user();
+        return isset($user) ? true : false;
```
Comment (KrzysztofDziedziechEscolasoft) [inline src/Http/Requests/ReportTermConsultationRequest.php:12]: isset return true or false why u make "isset($user) ? true : false;"
?

File: src/Models/ConsultationTerm.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    use HasFactory;
+
+    protected $fillable = [
+        'user_id',
+        'order_item_id',
+        'executed_at',
+        'executed_status',
+    ];
+
+    public function orderItem()
```
Comment (KrzysztofDziedziechEscolasoft) [inline src/Models/ConsultationTerm.php:20]:
```suggestion
    public function orderItem(): BelongsTo
```

File: src/Repositories/ConsultationTermsRepository.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    public function model(): string
+    {
+        return ConsultationTerm::class;
+    }
+
+    public function findByOrderItem(int $orderItemId): ConsultationTerm
+    {
+        return $this->model->newQuery()->where('order_item_id', '=', $orderItemId)->firstOrFail();
```
Comment (KrzysztofDziedziechEscolasoft) [inline src/Repositories/ConsultationTermsRepository.php:25]:
```suggestion
        return $this->model->newQuery()->where('order_item_id', $orderItemId)->firstOrFail();
```

File: src/Services/ConsultationService.php [hunk 4]
```diff
@@ -61,4 +67,33 @@ public function delete(int $id): ?bool
             return $this->consultationRepositoryContract->delete($id);
         });
     }
+
+    public function setPivotOrderConsultation($order, $user)
```
Comment (KrzysztofDziedziechEscolasoft) [inline src/Services/ConsultationService.php:71]:
```suggestion
    public function setPivotOrderConsultation($order, $user): void
```

File: tests/APIs/ConsultationReportTermTest.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                'total' => $price,
+                'subtotal' => $price,
+            ]);
+        Event::fake();
+        $event = new CartOrderPaid($this->user, $this->order);
+        $listener = app(ReportTermListener::class);
+        $listener->handle($event);
+    }
+
+    public function testConsultationReportTerm()
```
Comment (KrzysztofDziedziechEscolasoft) [inline tests/APIs/ConsultationReportTermTest.php:61]:
```suggestion
    public function testConsultationReportTerm(): void
```

add in other function in tests

**Review by qunabu — APPROVED** [review]

**Pull request merged**